src/main/java/dev/waterdog/waterdogpe/network/protocol/handler/downstream/ConnectedDownstreamHandler.java

```diff
@@ -39,18 +39,6 @@ public ConnectedDownstreamHandler(ProxiedPlayer player, ClientConnection connect
         super(player, connection);
     }
 
-    @Override
-    public PacketSignal handle(ItemComponentPacket packet) {
-        if (!this.player.acceptItemComponentPacket()) {
-            return Signals.CANCEL;
-        }
-        player.setAcceptItemComponentPacket(false);
-        if (this.player.getProtocol().isAfterOrEqual(ProtocolVersion.MINECRAFT_PE_1_21_60)) {
-            setItemDefinitions(packet.getItems());
-        }
-        return PacketSignal.UNHANDLED;
-    }
-
```

Comment:
For some reason sometimes proxy no longer switches quickly enough to handle this packet inside `ConnectedDownstreamHandler`, that is why it was moved to `AbstractDownstreamHandler`, so it can be handled in both `SwitchDownstreamHandler` and `ConnectedDownstreamHandler`